File: 1636-number-of-substrings-with-only-1s/number-of-substrings-with-only-1s.cpp
class Solution
{
public:
    int numSub(string s)
    {
        int mod = 1000000007;
        long count = 0;
        long ans = 0;
        for (int i = 0; i < s.size(); i++)
        {
            if (s[i] == '1')
            {
                count++;
            }
            else
            {
                long add = (count * (count + 1)) / 2;
                add = add % mod;
                ans = (ans + add) % mod;
                count = 0;
            }
        }
        long add = (count * (count + 1)) / 2;
        add = add % mod;
        ans = (ans + add) % mod;
        return (int)ans;
    }
};

File: 1636-number-of-substrings-with-only-1s/number-of-substrings-with-only-1s.java
class Solution {
    static final int M = 1_000_000_007;
    public int numSub(String s) {
        long result = 0;
        long count1 = 0;
        for(char ch : s.toCharArray()){
            if(ch == '1'){
                count1++;
                result = (result + count1) % M; 
            }else{
                count1 = 0;
            }
        }
        return (int) result;
    }
}